lib/Backend/LowerMDShared.cpp

```diff
@@ -8683,24 +8687,21 @@ void LowererMD::GenerateFastInlineBuiltInCall(IR::Instr* instr, IR::JnHelperMeth
             //     if (shouldCheckNegZero) {
             //         CMP roundedFloat, 0
             //         JA $setZero
-            //         JE $negZeroTest
-            //         J $bailout
+            //         JP $skipRoundSd (NaN)
+            //       $negZeroTest [Helper]:
+            //         JNE $bailoutLabel
+            //         isNegZero(src)
+            //         JE $bailoutLabel
+            //         J $skipRoundSd
             //     } else {
             //         CMP roundedFloat, 0
-            //         JL $bailout
-            //         J $setZero
-            //     }
+            //         JNE $setZero
```

Comment:
At least from my reading how `SetNegativeZeroDoesNotMatterIfLastUse()` is called, it looks more like we set that bit in case the result isn't used in a manner which would cause difference between +0 and -0 to be visible, not seeing any tracking of exactly the value -0.